### PR TITLE
Uploaded all files and fixed errors

### PR DESCRIPTION
**Pull request by DH3773**
Uploaded all of the theory files and fixed all of the syntax errors.

### CODE REVIEW

File: Theory/Note.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        return new Note(_allNotes[index]);
+    }
+
+    public Note SemitoneDown()
+    {
+        var index = Array.IndexOf(_allNotes, _letter);
+        
+        if(index == 0)
```
Comment (alyssa-blair) [inline Theory/Note.cs]: Everything looks good! Could update this to a modular calculation ((index + 11) % 12) if you wanted to but up to you!

File: Theory/Note.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            }
+        }
+    }
+
+
+    public Note SemitoneUp() 
+    {
+        var index = Array.IndexOf(_allNotes, _letter);
+
+        if (index == 11)
```
Comment (alyssa-blair) [inline Theory/Note.cs]: could also change this one to modular ((index + 1) % 12)

File: Theory/Chords.cs [hunk 3]
```diff
@@ -0,0 +1,366 @@
+namespace Fretty.Theory;
+
+public class Chords
```
Comment (wrighted) [inline Theory/Chords.cs]: This class can be `static` I think.

File: Theory/GuitarString.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+        for (int i = 1; i < _stringNotes.Length; i++)
+        {
+            _stringNotes[i] = _stringNotes[i - 1].SemitoneUp();
+        }
+    }
+
+    // Returns the note at the given fret
+    // where fret 0 is the open string 
+    public Note at_fret(int fret) {
```
Comment (wrighted) [inline Theory/GuitarString.cs]: `AtFret`

File: Theory/GuitarString.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        return _stringNotes[fret];
+    }
+
+    // Returns the array of Notes that 
+    // represents all of the notes on 
+    // the Guitar_String
+    public Note[] to_notes_array() {
```
Comment (wrighted) [inline Theory/GuitarString.cs]: `AsNotes`

File: Theory/Note.cs (hunk 2 above)
Comment (wrighted) [reply Theory/Note.cs]: Would prefer to see dynamic implementation:

```
int index = Array.IndexOf(_allNotes, _letter);
        
int newIndex = index >= _allNotes.Length - 1 ? 0 : index + 1;
return new Note(_allNotes[newIndex]);
```

File: Theory/Scales.cs [hunk 6]
```diff
@@ -0,0 +1,187 @@
+namespace Fretty.Theory;
+
+// Class holding all scales and scale-related functions
+public class Scales
```
Comment (wrighted) [inline Theory/Scales.cs]: This class can be `static`.

File: Theory/Scales.cs [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+    {
+        return AllScales[name];
+    }
+
+
+
+
+
+    //TODO, change this to use Chord class? Or maybe just have a function in chord class that returns array of notes
+    public static List<ScaleName> ScalesContainingChords(Note[][] chords)
```
Comment (wrighted) [inline Theory/Scales.cs]: I like the comments here, nice work.

File: Theory/Song.cs [hunk 8]
```diff
@@ -0,0 +1,30 @@
+namespace Fretty.Theory;
+
+public class Song(string songName, Key songKey, Chords.ChordName[] songChords)
```
Comment (wrighted) [inline Theory/Song.cs]: If you add:

`using Chords;`

above this type can become:

`ChordName[]`

File: Theory/Tuning.cs [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+    public Tuning(GuitarString[] tuning)
+    {
+        if(tuning.Length == 6) 
+        {
+            _strings = tuning;
+        }
+        else
+        {
+            //Throw exception
```
Comment (wrighted) [inline Theory/Tuning.cs]: I don't think this comment adds much.

File: Theory/Voicing.cs [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+public class Voicing
+{
+
+    // The name of the chord is dependent
+    // on the tuning, and so should be stored
+    // in a Tuning instance in a dictionary
+    // style, with {Chord: Voicing1, Voicing2, etc}
+    private int[] _positions;
+
+    public Voicing(int[] fretPositions, int[] positions)
```
Comment (wrighted) [inline Theory/Voicing.cs:20]: What's the intended use of a Voicing? ie) What is parameter `int[] positions`s role?

File: TheoryTests/NoteTest.cs [hunk 11]
```diff
@@ -0,0 +1,16 @@
+// namespace Fretty.TheoryTests;
```
Comment (wrighted) [inline TheoryTests/NoteTest.cs:1]: Still working on adding testing framework.

**Review by wrighted — COMMENTED** [review]
Just a few very minor changes I want to see, seems like a fairly robust object approach at representing guitar logic, nice work.

**Review by wrighted — APPROVED** [review]
Let's merge this so we can start testing end-to-end.

**Pull request merged**